### DISCUSSION TITLE
If create jobs with `scheduled_job` decorator, will jobs get re-created everytime?

### DISCUSSION THREAD

**Discussion by Toshinaki** [agronholm/apscheduler#714] [Q&A]
I'd like to use apscheduler with FastAPI, and defining jobs with `scheduled_job` decorator.

The schedule instance is created before app starts and shutdown after app ends. And I'm using postgres to store jobs.

If I instantiate apscheduler like this:
```python
from apscheduler.schedulers.asyncio import AsyncIOScheduler

scheduler = AsyncIOScheduler(...)
scheduler.start()

@scheduler.scheduled_job(...)
def job1():
    ...

@scheduler.scheduled_job(...)
def job2():
    ...
``` 

Every time the app restarts, will those jobs get added to the db? Or if jobs already exist, skip adding?

**Comment by agronholm** [comment]
If you supply an `id`, the new job will replace the old one, if it exists.

This decorator was a misfeature that has been removed in the 4.0 series (currently in alpha).

**Reply by Toshinaki** [reply]
So the recommended way to add jobs is `add_job()`?
If supply `id` to `add_job()`, and that `id` exists, the old job in DB will be replaced; if no `id` supplied, new (same) job will be created. 

To avoid that, I have to manually determine `id`s (or `name` maybe?) for all jobs, and on startup, check `id`s, if exists, skip.
Is that right?

**Reply by Toshinaki** [reply]
Also, do you think it's possible for schedulers to accept an yaml config file, to simplify this:
```python
scheduler = BackgroundScheduler({
    'apscheduler.jobstores.mongo': {
         'type': 'mongodb'
    },
    'apscheduler.jobstores.default': {
        'type': 'sqlalchemy',
        'url': 'sqlite:///jobs.sqlite'
    },
    'apscheduler.executors.default': {
        'class': 'apscheduler.executors.pool:ThreadPoolExecutor',
        'max_workers': '20'
    },
    'apscheduler.executors.processpool': {
        'type': 'processpool',
        'max_workers': '5'
    },
    'apscheduler.job_defaults.coalesce': 'false',
    'apscheduler.job_defaults.max_instances': '3',
    'apscheduler.timezone': 'UTC',
})
```
